src/snarks/groth16/mpc_utils.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef __ZETH_SNARKS_GROTH16_MPC_UTILS_HPP__
+#define __ZETH_SNARKS_GROTH16_MPC_UTILS_HPP__
+
+#include "snarks/groth16/powersoftau_utils.hpp"
+
+#include <vector>
+
+// Structures and utility functions related to CRS generation via an
+// MPC.
+
+namespace libzeth
+{
+
+using ppT = libff::default_ec_pp;
```

Comment:
The templates are "instantiated" in the `prover_server.cc` in order to make sure it is consistent everywhere. I can see that you have re-instantiated your templates in several files here. I'd move all these `using` directives to the program's entrypoint to ensure a consistent type handling (and I find it easier to reason about). We might want to keep the same template name and change the type with which we instantiate it for some exceptional cases, but here it does not seem to be one.

Reply:
Templatized the mpc code, removing this.